runtime/doc/various.txt

```diff
@@ -626,4 +626,62 @@ gc			Text object for the largest contiguous block of
 			`gcgc` uncomments a comment block; `dgc` deletes it).
 			Works only in Operator-pending mode.
 
+==============================================================================
+3. Matching							*matching*
+
+Nvim supports highlighting and jumping to matching groups based on |treesitter|
+nodes or |syntax| groups.
+
+The groups are highlighted with |hl-MatchParen|.
+
+							*matching-groups*
+Treesitter capture groups that count as groups:
+- `@punctuation.bracket`: Braces, brackets, and parentheses pairs.
+- `@keyword`: Language-specific keywords.
+
+Syntax supported matching groups:
+- 'matchpairs': Braces, brackets, and parentheses pairs.
+- |b:match_words|: Language-specific keywords.
+
+						*matching-%* *%-default*
+[count]%		Jump [count] matches forwards through the
+			|matching-groups|.
+
+						*matching-g%* *g%-default*
+[count]g%		Jump [count] matches backwards through the
+			|matching-groups|.
+
+CONFIGURATION					*matching-configuration*
+
+Extending the words recognized as matching groups is done using
+|b:match_words| or 'matchpairs'.
+
+							*b:match_words*
+The format for |b:match_words| is similar to 'matchpairs': a comma
+(,)-separated list of groups. Each group is a colon (:)-separated list of
+patterns |pattern|. Here's an example for some Lua keywords: >vim
+	:let b:match_words = '\<if\>:\<elseif\>:\<else\>:\<end\>,'
+		\ . '\<function\>:\<end\>'
+<
+Where |\<| and |\>| denote word boundaries. |b:match_words| also supports a
+|Funcref| as value: >vim
+	:let b:match_words = 'GetMatchWords()'
+<
+The group definitions also support a single backref group |\1|. For example,
+>vim
+	:let b:match_words = '\<\(if|function\)\>:\<end\1\>'
+<
+matches "if" with "endif" and "function" with "endfunction".
```

Comment:
Should there be a way to set this for multiple file types at a time? Otherwise people will need a bunch of `after/ftplugin/<lang>.lua` files to set `match_words`

Something more like how `vim.filetype.add()` is implemented?

Reply:
That is simply for backward compatibility with existing ftplugins. Going forward, it will not be advertised as a first-class configuration interface. (That is still TBD.)

Reply:
Should I keep it in the documentation? I wanted to keep the `b:match_words` tag for now.

Reply:
Yes, but it should state explicitly that these are supported for the sake of compatibility with `matchit`. Since we will keep shipping `matchit` (just not enable it by default), we can just link to their documentation and focus on our limitations (if any).